jupyter-on-gke/jupyterhub.tf

```diff
@@ -41,6 +41,14 @@ data "google_compute_backend_service" "jupyter-ingress" {
   project = var.project_id
 }
 
+resource "google_iap_web_backend_service_iam_binding" "binding" {
+  count = var.add_auth && data.google_compute_backend_service.jupyter-ingress.generated_id != null ? 1 : 0
```

Comment:
data.google_compute_backend_service.jupyter-ingress.generated_id != null --> why do we need this?

Reply:
What happens if the backend service doesn't exist yet? Does the terraform apply fail?

Reply:
Yup, it gives an 404 error in terraform saying that the service doesn't exist.